extensions/amp-viewer-integration/0.1/amp-viewer-integration.js

```diff
@@ -14,5 +14,71 @@
  * limitations under the License.
  */
 
+import {Messaging} from './messaging.js';
+import {listen} from '../../../src/event-helper';
+import {viewerForDoc} from '../../../src/viewer';
+import {user} from '../../../src/log';
+
+
 export class AmpViewerIntegration {
+
+  /**
+   * @param {!Window} win
+   */
+  constructor(win) {
+    /** @const @private {!Window} win */
+    this.win_ = win;
+  }
+
+  /**
+   * @return {!Promise}
+   */
+  init() {
+    return new Promise(resolve => {
```

Comment:
This can be written a bit more idiomatically like:

```js
init() {
  const viewer = viewerForDoc(this.win_.document);
  return this.getHandshakePromise_(viewer).then(viewerOrigin => {
    // messaging stuff
    // setMessageDeliverer stuff
  });
}